src/node_lkl.cc

```diff
@@ -150,3 +148,39 @@ NAN_METHOD(syscall) {
 		}
 	}
 }
+
+NAN_METHOD(parseDirent64) {
+	// Args:
+	// * Buffer buffer
+	// * int nread
+	// * Array result
+	// Parses buffer (which should be a lkl_linux_dirent64*) and fills the
+	// result array with filenames it finds.
+	// Filenames are Buffers that need to be decoded by the caller.
+	char* buf = (char*) node::Buffer::Data(info[0]);
+	unsigned int nread = info[1]->Uint32Value();
+	assert(nread <= node::Buffer::Length(info[0]));
```

Comment:
For the buffer you can easily slice the original buffer to the length specified by `nread`, but even if you don't do that you have to use safe string manipulation functions that specify the maximum amount of memory it is allowed to read. so instead of doing `strlen(d->name)` you can do `strnlen(dir_entry->d_name, max_length)`